dashboard/app/assets/javascripts/soundEffects.js

```diff
@@ -227,24 +227,35 @@ Sound.prototype.newPlayableBufferSource = function(buffer, options) {
   var newSound = this.audioContext.createBufferSource();
```

Comment:
Why sharing a player: IIRC there are some browser-specific crashes / playback failures you get if you initialize more than one audio player on a given `window`, so we initialize a single audio player and pass it in as a configuration parameter to apps / blockly-core. When phaser e.g. tries to initialize its own x-browser audio player separately, it crashes in certain browsers, so we used the shared one instead.

Why does dashboard need sounds: dashboard has to play sounds without `apps` initialized (on e.g. match/multi/etc levels for success/fail)

Why is the audio player in dashboard and not shared? It was made before shared existed, but I'm also not sure that putting it in shared and then only initializing it in dashboard would be much better?

Think you all know the rest of this context, but in case it helps:

Where'd the audio player logic come from? Originally stemmed from the playback logic in blockly-core, a version was copied into apps in 2013, cross browser bugs were fixed in the two versions in 2013/2014, then to avoid having to fix bugs in two places I joined the two implementations in late 2014 (at that time refactoring into the class that is there today but not making any functional changes to the browser playback logic).

What... why do we have a home rolled x-browser audio player library? Inertia! It may make sense to integrate an open source x-browser audio player that we confirm works across our matrix, esp. if it appears to fix a bug the current player doesn't (time-to-test might end up being net time savings in that case).